1.1/percentile.go

```diff
@@ -16,13 +18,26 @@ func createData(rand *rand.Rand, n uint) []uint {
 }
 
 type percentile func([]uint) uint
+type seedData []uint
+
+func (a seedData) Len() int { return len(a) }
+
+func (a seedData) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+
+func (a seedData) Less(i, j int) bool { return a[i] < a[j] }
 
 func main() {
 	var seed int64 = 1
 	rand := rand.New(rand.NewSource(seed))
 	var f percentile
-	// f = your function
-	data := createData(rand, 10000)
+	f = func(data []uint) uint {
+		length := len(data)
+		n := .9 * float32(length)
+		index := uint(math.Ceil(float64(n)))
+		return data[index]
+	}
+	data := seedData(createData(rand, 10000))
+	sort.Sort(data)
```

Comment:
Sorting needs to be part of your function.  You cannot assume input is ordered.

Also, it'd be good to keep in mind whether your percentile function should mutate the underlying data.  This will tie into part 2.